File: src/main/java/com/joecollins/graphics/screens/generic/PartyResult.java
package com.joecollins.graphics.screens.generic;

import com.joecollins.graphics.utils.ColorUtils;
import com.joecollins.models.general.Party;
import java.awt.Color;

public class PartyResult {

  private final Party party;
  private final boolean elected;

  public PartyResult(Party party, boolean elected) {
    this.party = party;
    this.elected = elected;
  }

  public static PartyResult elected(Party party) {
    return new PartyResult(party, true);
  }

  public static PartyResult leading(Party party) {
    return new PartyResult(party, false);
  }

  public Color getColor() {
    if (party == null) {
      return Color.BLACK;
    }
    if (elected) {
      return party.getColor();
    }
    return ColorUtils.lighten(party.getColor());
  }
}

File: models/src/main/kotlin/com/joecollins/models/general/PartyResult.kt
package com.joecollins.models.general

data class PartyResult(override val leader: Party, override val elected: Boolean) : ElectionResult<Party> {

    companion object {

        val TIE = PartyResult(Party.TIE, false)

        fun elected(party: Party?) = if (party == null) null else PartyResult(party, true)

        @JvmName("electedNotNull")
        fun elected(party: Party) = PartyResult(party, true)

        fun leading(party: Party?) = if (party == null) null else PartyResult(party, false)

        @JvmName("leadingNotNull")
        fun leading(party: Party) = PartyResult(party, false)
    }
}
